src/needleman_wunch.jl

```diff
@@ -1,15 +1,84 @@
 # Labs 3 & 4
 
 """
-    nwscore(base1, base2)
+    nwscore(base1, base2; match, mismatch, gap) --> Int
+
+
+    Accepts two bases and possibly reassigned match, mismatch, and gap scores, then returns the correct score based on the original Needleman Wunch paper.
+    If no bases are given, an error is thrown.
+
+    Examples
+    =========
+    nwscore(base, nothing) --> -1
+    nwscore(base, nothing; gap = -2) --> -2
+    nwscore('A', 'T'; mismatch=-2, match = 3, gap=-2) --> -2
 
-Your documentation here.
 """
-function nwscore(base1::Char, base2::Char)
-    
+function nwscore(base1::Char, base2::Char; match::Int = 1, mismatch::Int = -1, gap::Int = -1)
+    if base1 == base2
+        return match
+    elseif base1 != base2
+        return mismatch
+    end
+
 end
 
-nwscore(base::Char, ::Nothing) = -1
-nwscore(::Nothing, base::Char) = nwscore(base, nothing)
+function nwscore(base::Char, ::Nothing; match::Int = 1, mismatch::Int = -1, gap::Int = -1)
+    return gap
+end
+
+nwscore(::Nothing, base::Char;  match::Int = 1, mismatch::Int = -1, gap::Int = -1) = nwscore(base, nothing; match = match, mismatch = mismatch, gap = gap)
 
 nwscore(::Nothing, ::Nothing) = throw(ArgumentError("Score for two gaps is not defined"))
+nwscore(::Nothing, ::Nothing; match::Int = 1, mismatch::Int = -1, gap::Int = -1) = throw(ArgumentError("Score for two gaps is not defined"))
+
+
+function nwalign()
+    # aligner
+end
+
+function nwsetupmatrix(seq1::String, seq2::String; gap::Int = -1)
+    length1 = length(seq1)
+    length2 = length(seq2)
+    scoringMatrix = zeros(Int, length1+1, length2+1)
+
+    for i in 1:(length1+1)
```

Comment:
Kinda weird to nest these loops - imagine you have a sequence that's 1000 bases long, you're going to loop 999 times without doing anything. You could just do a loop with `i` set to 1, and loop through j, then do a different loop with `j` set to 1 and you loop through `i`.